src/winmain.cpp

```diff
@@ -590,6 +589,70 @@ LRESULT CALLBACK proxyDlgProc(HWND hWndDlg, UINT Msg, WPARAM wParam, LPARAM)
 	return FALSE;
 }
 
+struct UpdateCheckParams
+{
+	GupNativeLang& _nativeLang;
+	GupParameters& _gupParams;
+};
+
+LRESULT CALLBACK updateCheckDlgProc(HWND hWndDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+	switch (message)
+	{
+	case WM_INITDIALOG:
+	{
+		auto* params = reinterpret_cast<UpdateCheckParams*>(lParam);
+		if (params)
+		{
+			const string& title = params->_gupParams.getMessageBoxTitle();
+			if (!title.empty())
+				::SetWindowTextA(hWndDlg, title.c_str());
+			string textMsg = params->_nativeLang.getMessageString("MSGID_NOUPDATE");
+			if (!textMsg.empty())
+				::SetDlgItemTextA(hWndDlg, IDC_UPDATE_STATIC1, textMsg.c_str());
+			string textLink = params->_nativeLang.getMessageString("MSGID_DOWNLOADTEXT");
+			if (!textLink.empty())
+				::SetDlgItemTextA(hWndDlg, IDC_DOWNLOAD_LINK, textLink.c_str());
+		}
+		goToScreenCenter(hWndDlg);
+		return TRUE;
+	}
+
+	case WM_COMMAND:
+	{
+		switch LOWORD((wParam))
+		{
+		case IDOK:
+		case IDYES:
+		case IDNO:
+		case IDCANCEL:
+			EndDialog(hWndDlg, wParam);
+			return TRUE;
+		default:
+			break;
+		}
+	}
+
+	case WM_NOTIFY:
+		switch (((LPNMHDR)lParam)->code)
+		{
+		case NM_CLICK:
+		case NM_RETURN:
+		{
+			PNMLINK pNMLink = (PNMLINK)lParam;
+			LITEM item = pNMLink->item;
+			if (lstrcmpW(item.szID, L"id_download") == 0)
+			{
+				::ShellExecute(NULL, TEXT("open"), TEXT("https://notepad-plus-plus.org/downloads/"), NULL, NULL, SW_SHOWNORMAL);

```

Comment:
I didn't go into the debug mode, but the current line is not executed because `if (lstrcmpW(item.szID, L"id_download") == 0)`. I would change this line to `if (strcmpW(item.szID, L"id_download") == 0)`

Reply:
No, it was actually a typo in the localization XML.

Reply:
I was so tired that I took `l` as `!`.
Thank you for fixing it.

Reply:
@mere-human 
But this bug reveal one thing - if translator mess with the format while doing the translation, the download link won't work.

Reply:
@donho ok, should I remove the id=".." at all?
At first, I thought it might be a good idea that helps to distinguish between several link controls.
But now it seems redundant.

Reply:
I was about to delete that id="..." but then I saw this PR got merged.
Well then, another time, I guess 🙂